src/main/java/mate/academy/springboot/datajpa/service/mapper/ProductMapper.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.springboot.datajpa.service.mapper;
+
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMapper implements RequestDtoMapper<ProductRequestDto, Product>,
+        ResponseDtoMapper<Product, ProductResponseDto> {
+    private final CategoryService categoryService;
+
+    public ProductMapper(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    @Override
+    public Product mapToModel(ProductRequestDto productRequestDto) {
+        Product product = new Product();
+        product.setTitle(productRequestDto.getTitle());
+        product.setPrice(productRequestDto.getPrice());
+        product.setCategory(categoryService.findById(productRequestDto.getCategoryId()));
+        return product;
+    }
+
+    @Override
+    public ProductResponseDto mapToDto(Product product) {
+        ProductResponseDto productResponseDto = new ProductResponseDto();
+        productResponseDto.setId(product.getId());
+        productResponseDto.setTitle(product.getTitle());
+        productResponseDto.setPrice(product.getPrice());
+        productResponseDto.setCategoryId(product.getCategory().getId());
+        return productResponseDto;
+    }
```

Comment:
The same with other mappers. We already know that it is a mapper so `map` is redundant